client/allocrunner/taskrunner/template_hook.go

```diff
@@ -72,10 +73,12 @@ func (h *templateHook) Prestart(ctx context.Context, req *interfaces.TaskPrestar
 		return nil
 	}
 
+	l_restored, err := strconv.ParseBool(req.PreviousState["hoookRestoredFromRestart"])
```

Comment:
This isn't the best use of hook state I think a better approach would be to create a getter interface that the hook has a reference to that can be used here. We do similar things in other hooks that need to access task state or operations. For example take a look at the `StatsUpdater` interface that the stats hook uses:
https://github.com/hashicorp/nomad/blob/f02c163532238bb72baa3cd0df36e4143a3c909c/client/allocrunner/taskrunner/stats_hook.go#L15-L23

We can also easily mock out the interface when testing.